### ISSUE TITLE
Inconsistency in handling batches

### ISSUE THREAD

**Issue by ferrophile** [NVlabs/stylegan2-ada-pytorch#169]
Regarding the training procedure in `training_loop.py`, in line 281, a for loop loops over 4 tensors simulatenously to accumulate gradients for a single iteration, in each GPU:

`for round_idx, (real_img, real_c, gen_z, gen_c) in enumerate(zip(phase_real_img, phase_real_c, phase_gen_z, phase_gen_c)):`

However, I found that the 4 tensors have varying lengths in the batch dimension:
- `phase_real_img` has a length of `batch_size` // `num_gpus` // `batch_gpus`
    - Division by `num_gpus` comes from initialization of training set iterator (line 138, `batch_size=batch_size//num_gpus`)
    - Division by 'batch_gpus` comes from `split` operation in line 261
- `phase_real_c` has same length as above, by similar reasoning
- `phase_gen_z` has a length of `batch_size` // `batch_gpus`
    - Division by `batch_gpus` comes from `split` operation in line 264
- `phase_gen_c` has same length as above, by similar reasoning

In short, during the aforementioned for loop, `phase_gen_z` and `phase_gen_c` has N times the length of `phase_real_img` and `phase_real_c`, where N is the no. of GPUs.

More specifically, when I used 4 GPUs to train and set `batch_size=32`, `batch_gpu=4`, `phase_real_img` / `phase_real_c` has a length of 2, and `phase_gen_z` / `phase_gen_c` has a length of 8. `zip` should generate up to the smaller count (2 in this case) so the remaining 6 entries for  `phase_gen_z` / `phase_gen_c` are wasted.

In theory this has no effect on the training result, but I still find the code a bit weird. Is this result intentional or not?

**Comment by pranurs** [comment]
I also stumbled upon this today, is there any explanation for this, or should we divide the batch_size by num_gpus as mentioned by @gauku in the previous comment?
